### PR TITLE
PLT-4098 Don't show @mention suggestions if word chars immediately precede at sign

### PR DESCRIPTION
**Pull request by thomchop**
#### Summary
If an at sign is immediately preceded by one or more word characters in the text input for a post, the user is most likely typing an email address and it doesn't make sense to show the at-mention suggestion box.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-4098

#### Checklist
- [x] Has UI changes

### CODE REVIEW

**Comment by jwilander** [comment]
Closing/re-opening to trigger build

**Comment by coreyhulen** [comment]
yeah...this is one for @hmhealey to review :)

File: webapp/components/suggestion/at_mention_provider.jsx [hunk 1]
```diff
@@ -112,9 +112,10 @@ export default class AtMentionProvider {
     handlePretextChanged(suggestionId, pretext) {
         clearTimeout(this.timeoutId);
 
-        const captured = (/@([a-z0-9\-\._]*)$/i).exec(pretext.toLowerCase());
-        if (captured) {
-            const prefix = captured[1];
+        const captured = (/(\w*)@([a-z0-9\-\._]*)$/i).exec(pretext.toLowerCase());
```
Comment (hmhealey) [inline webapp/components/suggestion/at_mention_provider.jsx]: You can actually make this a bit simpler by changing the regex to `/(?:^|\W)@([a-z0-9\-\/_]*)*/i`.  That way, it's a bit clearer that it's looking for an at mention that's either at the start of the text or following any non-word character, and the `?:` will mean the `const prefix = captured[1];` line below won't need to change.

Comment (thomchop) [inline webapp/components/suggestion/at_mention_provider.jsx]: 👍  I'll make that change.

Comment (thomchop) [inline webapp/components/suggestion/at_mention_provider.jsx]: Ok no prob. I came up with a workaround.

Comment (hmhealey) [inline webapp/components/suggestion/at_mention_provider.jsx]: Oh right. I forgot the first group is everything matched, even if the group isn't included. I used `/(^|\W)(@([a-z0-9.\-_]*))/` in text_formatting.jsx, but your way works just as well. Thanks for changing that!

**Pull request merged**